requirements.txt

```diff
@@ -0,0 +1,20 @@
+boto3==1.26.22
+botocore==1.29.22
+certifi==2022.9.24
+charset-normalizer==2.1.1
+click==8.1.3
+Deprecated==1.2.13
+idna==3.4
+importlib-metadata==5.1.0
+jmespath==1.0.1
+keyring==23.4.1
+python-dateutil==2.8.2
+PyYAML==6.0
+requests==2.28.1
+s3transfer==0.6.0
+six==1.16.0
+synapseclient==2.7.0
+synapseformation==0.0.3
+urllib3==1.26.13
+wrapt==1.14.1
+zipp==3.11.0
```

Comment:
requirements.txt only needs the dependencies, not the dependencies dependencies etc. And it doesn't need anything from the [python standard library](https://docs.python.org/3/library/) since those come with python.

Apparently [there are tools that automate this process for you](https://github.com/damnever/pigar), although I've never used them so I can't vouch for them. Usually I look at the imports and put anything that's not already in the  python standard library in requirements.txt.

In this case that's just [these libraries](https://github.com/Sage-Bionetworks/bridge-analytics-template/blob/main/src/copy_from_template.py#L27-L31), although you can omit `synapseutils` since that's installed automatically alongside `synapseclient`.